### ISSUE TITLE
[a11y]: Closing Modals when focus is on Search component

### ISSUE THREAD

**Issue by McOwska** [carbon-design-system/carbon#18045]
### Package

@carbon/react

### Package version

1.63.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Description

Closing Modals (as well as Tearsheets from @carbon/ibm-products) with the ESC key doesn't work when the focus is set on "Search type" component (like for example Search or Combobox) even when component's dropdown is closed and search field empty. When focus is on any other component within the Modal (including TextInputs) closing with ESC works fine. 

### Reproduction/example

https://stackblitz.com/edit/github-x5djhj?file=src%2FApp.jsx

### Steps to reproduce

1. Open the modal/tearsheet with the "Open Modal"/"Open Tearsheet" button
2. Navigate with a keyboard to the Search component
3. Try to close the modal by pressing the ESC key

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
Hey @McOwska 
This is the expected default behaviour for the Search component.  The [ event.stopPropagation ](https://github.com/carbon-design-system/carbon/blob/f2ba3fd9912ffdb96e672064364dd869a91c9d51/packages/react/src/components/Search/Search.tsx#L205) is intentional on the `handleKeyDown` function for `the Escape` key

_Escape Key Behavior in Search Component:_
Input Clearing Mechanism:
  - When Escape is pressed with text in the input, the component clears the input
  - This is a standard UX pattern allowing quick text removal
  - Prevents unnecessary typing or manual deletion

Event Propagation Risks Without `stopPropagation()`:
- Escape key would bubble up through parent components
- Modal's `onRequestClose` method could be accidentally triggered
- Modal might close when user only intends to clear search input

To achieve what you are expecting, you could pass onKeyDown handler on `Search` : [STACKBLITZ example](https://stackblitz.com/edit/github-x5djhj-cdj2aa?file=src%2FApp.jsx)

```
<Modal open={isModalOpen} onRequestClose={() => setIsModalOpen(false)}>
  <Search 
    onKeyDown={(e) => {
      if (e.key === 'Escape' && e.target.value === '') {
        setIsModalOpen(false);
      }
    }}
  />
</Modal>
```

**Comment by 2nikhiltom** [comment]
Closing this with the pervious comment
